### ISSUE TITLE
Credo.Check.Warning.WrongTestFileExtension false positive when using `--read-from-stdin`

### ISSUE THREAD

**Issue by rami3l** [rrrene/credo#1207]
## Background

When using the `credo` diagnostic source in `none-ls.nvim`, I realized this source [uses the `mix credo suggest --read-from-stdin` command](https://github.com/nvimtools/none-ls.nvim/blob/db2a48b79cfcdab8baa5d3f37f21c78b6705c62e/lua/null-ls/builtins/diagnostics/credo.lua#L37). However, when I enter `*_test.exs` files that are **indirect children** of the `./test/` directory, I will start to observe `Credo.Check.Warning.WrongTestFileExtension` false positives.

It has been mentioned in https://github.com/pantajoe/vscode-elixir-credo/issues/464#issue-1982508240 that passing a file name might help `credo` figure out whether to trigger file name-related lints, however being new to Elixir I'm not completely sure how that is working (or is supposed to work) in Credo.

## Problem Description

Given the following file:

```console
> cat ./test/angela_test.exs 
defmodule AngelaTest do
  use ExUnit.Case
  doctest Angela

  test "greets the world" do
    assert Angela.hello() == :world
  end
end
```

... `mix credo` provides no suggestion in this case, which is expected:

```console
> mix credo suggest --read-from-stdin ./test/angela_test.exs < ./test/angela_test.exs 
Checking 1 source file ...
[..]
1 mods/funs, found no issues.
```

... however, if I slightly change the file name, the false positive will occur (note that the file name still ends with `_text.exs`):

```console
> mix credo suggest --read-from-stdin ./test/angela/foo_test.exs < ./test/angela_test.exs
Checking 1 source file ...

  Warnings - please take a look                                           
┃ 
┃ [W] ↗ Test files should end with `_test.exs`.
┃       test/angela/foo_test.exs:1 #(AngelaTest)
[..]
1 mods/funs, found 1 warning.
```

It should be noted that the above reproduction is for the sake of convenience. The false positive persists if I have an actual `./test/angela/foo_test.exs` with the same contents.

## Environment

The host environment is shown as follows:

```console
# Credo version
> mix credo --version
1.7.12
```

```console
# Erlang & Elixir versions
> elixir --version
Erlang/OTP 28 [erts-16.0.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit] [dtrace]

Elixir 1.18.4 (compiled with Erlang/OTP 27)
```

```console
# OS version
> sw_vers 
ProductName:            macOS
ProductVersion:         15.5
BuildVersion:           24F74
```

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It should be fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by rami3l** [comment]
> Thanks for reporting this 😀 It should be fixed on `master`.
> 
> You can try this by setting the Credo dep to
> 
> {:credo, github: "rrrene/credo"}
> Please report back if your issue is solved! 👍

Many thanks for your prompt reply!

After updating to `master` the issue seems to have indeed been addressed on my machine.
As such, I am closing this issue as resolved by https://github.com/rrrene/credo/commit/c43dd583f6b46af7ec9d5f76041cfde7f80b76b4.
